packages/legend-application-query/src/components/QueryBuilderExplorerPanel.tsx

```diff
@@ -113,6 +116,48 @@ const checkForDeprecatedNode = (
   return false;
 };
 
+const isNodeAlreadyUsed = (
+  node: QueryBuilderExplorerTreeNodeData,
+  projectionState: QueryBuilderProjectionState,
+): boolean => {
+  if (
+    node.id === '@dummy_rootNode' &&
```

Comment:
either we publish the root node ID as a constant and use it here, or check for root node using some other means, i.e. check that it doesn't have a parentNode?